src/test/scala/beam/sim/SecondaryActivitiesSpec.scala

```diff
@@ -0,0 +1,171 @@
+import scala.collection.mutable
+import scala.collection.mutable.ListBuffer
+import scala.io.Source
+
+class SecondaryActivitiesSpec
+    extends WordSpecLike
+    with Matchers
+    with BeamHelper
+    with GenericEventsSpec
+    with BeforeAndAfterAll {
+
+  def config: com.typesafe.config.Config =
+    ConfigFactory
+      .parseString(s"""
+                    |beam.agentsim.lastIteration = 0
+                    |beam.exchange.scenario.urbansim.activitySimEnabled = true
+                    |beam.agentsim.agents.tripBehaviors.mulitnomialLogit.generate_secondary_activities = true
+                    |beam.agentsim.agents.tripBehaviors.mulitnomialLogit.intercept_file_path = "test/test-resources/beam/agentsim/activities/activity-intercepts.csv"
+                    |beam.agentsim.agents.tripBehaviors.mulitnomialLogit.activity_file_path = "test/test-resources/beam/agentsim/activities/activity-params.csv"
+                     """.stripMargin)
+      .withFallback(testConfig("test/input/beamville/beam.conf"))
+      .resolve()
+
+  private var injector: inject.Injector = _
+  private var interceptMode: Set[String] = _
+
+  override def beforeAll(): Unit = {
+    val beamConfig = BeamConfig(config)
+    val beamScenario = loadScenario(beamConfig)
+    val configBuilder = new MatSimBeamConfigBuilder(config)
+    val matsimConfig = configBuilder.buildMatSimConf()
+    matsimConfig.planCalcScore().setMemorizingExperiencedPlans(true)
+    FileUtils.setConfigOutputFile(beamConfig, matsimConfig)
+
+    val scenario = ScenarioUtils.loadScenario(matsimConfig).asInstanceOf[MutableScenario]
+    scenario.setNetwork(beamScenario.network)
+
+    injector = org.matsim.core.controler.Injector.createInjector(
+      matsimConfig,
+      module(config, beamConfig, scenario, beamScenario)
+    )
+
+    beamServices = injector.getInstance(classOf[BeamServices])
+
+    val popAdjustment = DefaultPopulationAdjustment(beamServices)
+    popAdjustment.update(scenario)
+
+    eventManager = injector.getInstance(classOf[EventsManager])
+    networkHelper = injector.getInstance(classOf[NetworkHelper])
+
+    val interceptFilePath = beamConfig.beam.agentsim.agents.tripBehaviors.mulitnomialLogit.intercept_file_path
+    val interceptPath = Source.fromFile(interceptFilePath)
+    interceptMode = interceptPath.getLines().next().split(",").drop(1).toSet
+    interceptPath.close()
+  }
+
+  "Beam Run" should {
+
+    "check increased activities " in {
+      val eventsType = new ListBuffer[String]()
+      val basicEventHandler = new BasicEventHandler {
+        override def handleEvent(event: Event): Unit = {
+          event match {
+            case event: ActivityEndEvent =>
+              eventsType.append(event.getActType)
+            case _ =>
+          }
+        }
+      }
+      processHandlers(List(basicEventHandler))
+      assert(eventsType.toSet.intersect(interceptMode) == interceptMode)
+    }
+
+    "check mode for secondary activities" in {
```

Comment:
nitpick: more comments/words what is checked?

Reply:
rename this test case